### DISCUSSION TITLE
Adding Ethernet to an STM32 design

### DISCUSSION THREAD

**Discussion by ITstreet1** [stm32duino/Arduino_Core_STM32#2195] [General]
Hi guys,

Can anyone explain to me a thing or two how to add an Ethernet to an STM32 IC? Specifically the STM32F407VGT6.
According to the DS, this IC has an Ethernet.
https://datasheet.lcsc.com/lcsc/2304140030_STMicroelectronics-STM32F407VGT6_C12345.pdf

I have never designed an STM32 with Ethernet, so I need some guidance. I did design it with the W5500.

One more. Can this STM32 with Ethernet be programmed with the Arduino IDE? If so, what lib?
Some light here would be nice.

**Comment by fpistm** [comment]
Hi @ITstreet1 
Never do this too.
You can probably have a look to some schematics. Example the one for[ Nucleo 144 F429ZI](https://www.st.com/resource/en/schematic_pack/nucleo_144pins_sch.zip).

![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/20641798/a269ea95-ff9e-4d06-80fc-13feb04d148b)


Note that you will need an externa PHY as stated in the product specification:
![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/20641798/f8f104ae-7eee-4426-862b-64ae918c058b)


About the library:
https://github.com/stm32duino/STM32Ethernet

**Comment by fpistm** [comment]
Closed this one as linked to https://github.com/orgs/stm32duino/discussions/2197
